src/pages/comparison/comparison.tsx

```diff
@@ -175,11 +196,32 @@ const RightLayersPanelWrapper = styled(LeftLayersPanelWrapper)`
   })};
 `;
 
+const getLayerExamples = (): LayerExample[] => Object.values(EXAMPLES);
+
 export const Comparison = ({ mode }: ComparisonPageProps) => {
   let currentViewport: WebMercatorViewport = null;
+
+  const [leftSideExamples, setLeftSideExamples] = useState<LayerExample[]>(() =>
+    getLayerExamples()
+  );
+
+  const [rightSideExamples, setRightSideExamples] = useState<LayerExample[]>(
+    () => getLayerExamples()
+  );
+
+  const [leftSideSelectedExampleId, setLeftSideSelectedExampleId] =
+    useState<string>("");
+
+  const [rightSideSelectedExampleId, setRightSideSelectedExampleId] =
+    useState<string>("");
+
+  const [baseMaps, setBaseMaps] = useState<BaseMap[]>(BASE_MAPS);
+  const [selectedBaseMapId, setSelectedBaseMapId] = useState<string>("Dark");
+  const [selectedMapStyle, setSelectedMapStyle] = useState(INITIAL_MAP_STYLE);
```

Comment:
Can you track `selectedBaseMap` object that contains both `id` and `style`?

Reply:
Refactored